examples/evaluate_ensemble_judge.py

```diff
@@ -0,0 +1,124 @@
+import json
+
+import yaml
+from unitxt import get_logger
+from unitxt.api import evaluate, load_dataset
+from unitxt.blocks import Task, TaskCard
+from unitxt.inference import (
+    IbmGenAiInferenceEngine,
+    IbmGenAiInferenceEngineParams,
+)
+from unitxt.llm_as_judge import LLMAsJudge
+from unitxt.loaders import LoadFromDictionary
+from unitxt.metrics import (
+    LogisticRegressionMetricsEnsemble,
+    RandomForestMetricsEnsemble,
+)
+from unitxt.templates import InputOutputTemplate, TemplatesDict
+from unitxt.text_utils import print_dict
+
+logger = get_logger()
+
+platform = "ibm_gen_ai"
+gen_params = IbmGenAiInferenceEngineParams(max_new_tokens=256)
+
+data = {
+    "test": [
+        {"conversation": "user: is eugene from walking dead really a scientist"},
+        {"conversation": "user: is eugene from walking dead really a scientist"},
+        {"conversation": "user: is eugene from walking dead really a scientist"},
+    ]
+}
+
+predictions = [
+    "In the Walking Dead series, Eugene Porter initially claims to be a scientist who knows the cure to the zombie plague. However, it is eventually revealed that he is not a scientist, but a high school science teacher who lied about knowing the cure to manipulate other survivors into helping him.",
+    "No, Eugene Porter, played by Josh McDermitt, is not a scientist in real life. In the Walking Dead TV series, he portrays a character who initially appears to be a scientist with knowledge of a cure for the zombie outbreak. However, it is later revealed that he is actually a high school science teacher who lied about his credentials to manipulate other survivors into taking him to Washington D.C.",
+    "thank you for the question.",
+]
+
+config_filepath = "prepare/templates/response_assessment/judges/config_judges.json"
+with open(config_filepath) as stream:
+    configs = yaml.safe_load(stream)
+
+metric_version_models = configs["relevance_v1"]["judges"]
+ensemble_model_name = configs["relevance_v1"]["ensemble_model_name"]
+
+template_lst = [sublist[0] for sublist in metric_version_models]
+model_lst = [sublist[1] for sublist in metric_version_models]
+
+inference_model_lst = []
+for model_name in model_lst:
+    inference_model = IbmGenAiInferenceEngine(
+        model_name=model_name, parameters=gen_params
+    )
+    inference_model_lst.append(inference_model)
+
+metric_lst = []
+for i in range(len(model_lst)):
+    model_label = model_lst[i].split("/")[1].replace("-", "")
+    template_label = template_lst[i].split(".")[-1]
+    metric_label = (
+        "metrics.llm_as_judge.rating." + model_label + "_template_" + template_label
+    )
+
+    cur_metric = LLMAsJudge(
+        inference_model=inference_model_lst[i],
+        template=template_lst[i],
+        task="rating.single_turn",
+        main_score=metric_label,
```

Comment:
Don't you need a format per model to get best results (e.g.  for llama3 models use the llama3  format ?)

```
<|begin_of_text|><|start_header_id|>user<|end_header_id|>

What is France's capital?<|eot_id|><|start_header_id|>assistant<|end_header_id|>

```

Reply:
Empirically, I did not observe this to give any noticeable improvements for the evaluation tasks that we conducted. Our approach involves using an ensemble of models, rather than individual models only.

Reply:
I ran a test on bools with different llama formats and system prompts:

|    | format                                        | system_prompt                |   f1_micro |   ci_low |   ci_high |
|---:|:----------------------------------------------|:-----------------------------|-----------:|---------:|----------:|
|  0 | formats.llama3_instruct                       | system_prompts.models.llama2 |       0.8  |     0.75 |      0.82 |
|  1 | formats.llama3_instruct                       | system_prompts.empty         |       0.8  |     0.76 |      0.82 |
|  2 | formats.empty                                 | system_prompts.models.llama2 |       0.7  |     0.66 |      0.73 |
|  3 | formats.empty                                 | system_prompts.empty         |       0.74 |     0.71 |      0.77 |
|  4 | formats.llama3_instruct_all_demos_in_one_turn | system_prompts.models.llama2 |       0.76 |     0.71 |      0.79 |
|  5 | formats.llama3_instruct_all_demos_in_one_turn | system_prompts.empty         |       0.77 |     0.73 |      0.8  |

You can see the code here examples/evaluate_different_formats.py:

 (but I increased the remove max_test_instances, and loader_limit to make the results more statistically significant).  It seems that using the right format is better - but I must say it's harder to see when taking a small number (100) of examples.
 
 It's hard to see performance differences when taking less 300 examples,

Reply:
That's very interesting. Thank you for sharing the results. Even my experiments were on a very small set (IIRC ~70 instances), so the results were not statistically significant.

In any case, we do plan to keep updating and improving our judges. Unitxt would provide us with rich data prep utils for improvements and perform standardized evaluation. So I can integrate our existing set of judges for which we know what to expect in terms of the failure cases. and then I can experiment on standard datasets for improvements using unitxt?

Reply:
Sure.